proxy/logging/unit-tests/test_LogUtils.cc

```diff
@@ -153,3 +201,57 @@ TEST_CASE("get_unrolled_filename parses possible log files as expected", "[get_u
   constexpr ts::TextView no_dot = "logging_yaml";
   REQUIRE(get_unrolled_filename(no_dot) == no_dot);
 }
+
+TEST_CASE("LogUtils pure escapify url", "[pure_esc_url]")
+{
+  char input[][32] = {
+    " ",
+    "%",
+    "% ",
+    "%20",
+  };
+  const char *expected[] = {
+    "%20",
+    "%25",
+    "%25%20",
+    "%2520",
+  };
+  char output[128];
+  int output_len;
+
+  int n = sizeof(input) / sizeof(input[0]);
+  for (int i = 0; i < n; ++i) {
+    LogUtils::pure_escapify_url(NULL, input[i], std::strlen(input[i]), &output_len, output, 128);
+    if (std::strcmp(output, expected[i]) != 0) {
```

Comment:
`CHECK_THAT(output, Catch::Equals(expected[i]));` is better?

Reply:
Yes if you prefer.

Reply:
Hmm the Catch documentation says that the parameter to Equals must be a literal.

Reply:
Changed it to a simple CHECK instead.

Reply:
I thought `Equals` of String matcher could be used. But not a big deal. Simple `CHECK` looks fine.
https://github.com/catchorg/Catch2/blob/master/docs/matchers.md#string-matchers

Reply:
I have not tried but this is what https://github.com/catchorg/Catch2/blob/master/docs/matchers.md#string-matchers seems to be saying the parameter to Equals must be a literal string. "The string matchers are StartsWith, EndsWith, Contains, Equals and Matches. The first four match a literal (sub)string against a result...".

Reply:
Ah, OK. I missed the sentence. Thanks.